### PR TITLE
[metrics-profiling PART 2] add metrics and profiling tools 

### PR DESCRIPTION
**Pull request by Popcore**
This PR installs Prometheus and the pprof package to allow for the capturing of metrics and profiling information.

Metrics are exposed at a port different than the one used by the application server and the actual port number can be configured using the `--mport `  or `-m` flag.

Debugging profile can be activated passing the `--pprof` or `-r` flag when starting the server. This will mount additional handlers at /debug/pprof/... that can be used with the pprof go tool.

### CODE REVIEW

File: indexers/providers/flower_power.go [hunk 1]
```diff
@@ -31,12 +33,24 @@ const (
 	// maxConcurrency is upper limit of the number of concurrent jobs that can
 	// happen. It is used here to control how many Things the indexer will
 	// fetch in parallel.
-	maxConcurrency = 10
+	maxConcurrency = 5
```
Comment (chespinoza) [inline indexers/providers/flower_power.go]: The only doubt I have about this PR is here, why are we limiting the http requests?
if for some reason it is needed why not parameterized the value as a executable's flag?

File: datastore/user_datastore.go [hunk 2]
```diff
@@ -220,6 +233,9 @@ func (u UserStore) DeleteUser(userUID string) (err error) {
 		return errors.ErrUserNotFound
 	}
 
+	// usersMetric.With(prometheus.Labels{"action": "deleted"}).Inc()
```
Comment (smulube) [inline datastore/user_datastore.go]: is this comment to be removed?

File: cmd/server.go [hunk 3]
```diff
@@ -46,20 +63,22 @@ func ServerCmd() *cobra.Command {
 
 			db.DB.SetMaxOpenConns(5)
```
Comment (smulube) [inline cmd/server.go]: as with the @chespinoza 's other comment - this value should maybe be a command line configurable option. I think both changes maybe belong in a separate PR though

File: indexers/providers/flower_power.go [hunk 4]
```diff
@@ -128,9 +142,19 @@ func (f FlowerPower) IndexAllThings(indexContinuously bool) error {
 				f.Config.Logger.LogFromMap(buildLogMsg(indxErr.url, "error", indxErr.err.Error()))
 			}
 
+			for _, t := range things {
+				indexerStatusMetric.With(prometheus.Labels{
+					"resourceURL": t.Endpoint.URL,
```
Comment (smulube) [inline indexers/providers/flower_power.go]: so here the indexerStatusMetric is partitioned by `resourceURL` and `status`. Is `resourceURL` an individual thing? If that is the case I don't think you want to do that - the cardinality would be too high I believe. The prometheus docs states: 

> CAUTION: Remember that every unique combination of key-value label pairs represents a new time series, which can dramatically increase the amount of data stored. Do not use labels to store dimensions with high cardinality (many different label values), such as user IDs, email addresses, or other unbounded sets of values.

maybe we should rethink this, and think about why you wanted to partition like this, i.e. is there some other way of dividing up indexerStatus that would be more useful?

File: cmd/server.go (hunk 3 above)
Comment (Popcore) [reply cmd/server.go]: I created an issue for making these constants configurable. I will address it in a separate PR

File: indexers/providers/flower_power.go (hunk 1 above)
Comment (Popcore) [reply indexers/providers/flower_power.go]: I have created [issue#39](created https://github.com/thingful/grow/issues/39) to address this

File: glide.yaml [hunk 5]
```diff
@@ -27,3 +27,9 @@ import:
   subpackages:
   - log
 - package: github.com/pinzolo/casee
+- package: github.com/prometheus/client_golang
+  version: v0.8.0
+  subpackages:
+  - prometheus
+- package: github.com/prometheus/prometheus
```
Comment (smulube) [inline glide.yaml]: do we use this package anywhere?

File: datastore/user_datastore.go (hunk 2 above)
Comment (Popcore) [reply datastore/user_datastore.go]: Ops yeah I forgot to push the update removing this line. It's done now

File: glide.yaml (hunk 5 above)
Comment (Popcore) [reply glide.yaml]: I'm using it it when creating prometheus metrics in the metrics package:
https://github.com/thingful/grow/blob/metrics-and-profiling-2/metrics/metrics.go

Comment (Popcore) [reply glide.yaml]: Uh actually no, I'm importing github.com/prometheus/client_golang/prometheus. I'll try to remove it and run glide again

Comment (Popcore) [reply glide.yaml]: @smulube Removed
